lib/tests/unit/external_module_test.cc

```diff
@@ -209,30 +209,134 @@ TEST_CASE("ExternalModule::callAction - blocking", "[modules]") {
+    SECTION("succeeds when presented with a valid configuration") {
+        ExternalModule e_m { PXP_AGENT_ROOT_PATH
+                             "/lib/tests/resources/modules/convert_test"
+                             EXTENSION,
+                             lth_jc::JsonContainer(
+                               "{ \"rate\" : 1.1, \"fee_percent\" : 0.1, \"fee_max\" : 10 }"
+                             ) };
+        REQUIRE_NOTHROW(e_m.validateConfiguration());
+    }
+
+    // TEST: handles invalid configuration
+    SECTION("throws exception when presented with an invalid configuration") {
+        ExternalModule e_m { PXP_AGENT_ROOT_PATH
+                             "/lib/tests/resources/modules/convert_test"
+                             EXTENSION,
+                             lth_jc::JsonContainer(
+                               "{ \"rate\" : 1.1, \"fee_percent\" : 0.1, \"foo_max\" : 10 }"
+                             ) };
+        REQUIRE_THROWS_AS(e_m.validateConfiguration(),
+                          PCPClient::validation_error);
+    }
+}
+
+TEST_CASE("ExternalModule::executeAction", "[modules][output]") {
+    configureTest();
+    lth_util::scope_exit config_cleaner { resetTest };
+
+    // TEST: gets outcome
+    SECTION("gets the action output") {
+        ExternalModule e_m { PXP_AGENT_ROOT_PATH
+                             "/lib/tests/resources/modules/convert_test"
+                             EXTENSION,
+                             lth_jc::JsonContainer(
+                               "{ \"rate\" : 1.1, \"fee_percent\" : 0.1, \"fee_max\" : 10 }"
+                             ) };
+        std::string convert_txt {
+            (DATA_FORMAT % "\"0632\""
+                         % "\"convert_test\""
+                         % "\"convert\""
+                         % "{\"amount\" : 1000}").str() };
+        PCPClient::ParsedChunks convert_content {
+            lth_jc::JsonContainer(ENVELOPE_TXT),
+            lth_jc::JsonContainer(convert_txt),
+            NO_DEBUG,
+            0 };
+        ActionRequest request { RequestType::Blocking, convert_content };
 
         REQUIRE_NOTHROW(e_m.executeAction(request));
-        REQUIRE(fs::exists(pid_path));
+        REQUIRE(e_m.executeAction(request).results.get<double>("amount") == 1098.9);
+    }
 
-        try {
-            auto pid_txt = lth_file::read(pid_path.string());
-            auto pid = std::stoi(pid_txt);
-        } catch (std::exception) {
-            FAIL("fail to get pid");
-        }
+    // TEST: validates outcome
+    SECTION("validates the action output") {
+        ExternalModule e_m { PXP_AGENT_ROOT_PATH
+                             "/lib/tests/resources/modules/convert_test"
+                             EXTENSION,
+                             lth_jc::JsonContainer(
+                               "{ \"rate\" : 1.1, \"fee_percent\" : 0.1, \"fee_max\" : 10 }"
+                             ) };
+        std::string convert_txt {
```

Comment:
I'm getting different errors when I try to use auto. In the simplest case, for the following code:
```C++
auto convert_txt {
  (DATA_FORMAT % "\"0632\""
               % "\"convert_test\""
               % "\"convert\""
               % "{\"amount\" : 1000}").str() }
```
I'm getting:
```
/Users/mruzicka/git/pxp-agent/lib/tests/unit/external_module_test.cc:352:13: error: no matching conversion for functional-style cast
      from 'std::initializer_list<std::__1::basic_string<char> >' to 'lth_jc::JsonContainer'
            lth_jc::JsonContainer(convert_txt),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```